chains/manager.go

```diff
@@ -1569,3 +1592,238 @@ func (m *manager) getOrMakeVMGatherer(vmID ids.ID) (metrics.MultiGatherer, error
 	m.vmGatherer[vmID] = vmGatherer
 	return vmGatherer, nil
 }
+
+// createHandler creates a handler that passes messages from the network to the consensus engine
+func (m *manager) createHandler(ctx *snow.ConsensusContext, vm *block.ChangeNotifier, sb subnets.Subnet, primaryAlias string, connectedValidators tracker.Peers, peerTracker *p2p.PeerTracker, halter common.Halter) (handler.Handler, error) {
+	handlerReg, err := metrics.MakeAndRegister(
+		m.handlerGatherer,
+		primaryAlias,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	// Asynchronously passes messages from the network to the consensus engine
+	return handler.New(
+		ctx,
+		vm,
+		vm.WaitForEvent,
+		m.Validators,
+		m.FrontierPollFrequency,
+		m.ConsensusAppConcurrency,
+		m.ResourceTracker,
+		sb,
+		connectedValidators,
+		peerTracker,
+		handlerReg,
+		halter.Halt,
+	)
+}
+
+// createMessageSender creates a sender that passes messages from the consensus engine to the network
+func (m *manager) createMessageSender(ctx *snow.ConsensusContext, sb subnets.Subnet) (common.Sender, error) {
+	msgSender, err := sender.New(
+		ctx,
+		m.MsgCreator,
+		m.Net,
+		m.ManagerConfig.Router,
+		m.TimeoutManager,
+		p2ppb.EngineType_ENGINE_TYPE_CHAIN,
+		sb,
+		ctx.Registerer,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't initialize sender: %w", err)
+	}
+
+	if m.TracingEnabled {
+		msgSender = sender.Trace(msgSender, m.Tracer)
+	}
+
+	return msgSender, nil
+}
+
+func (m *manager) createTrackedPeers(primaryAlias string) (tracker.Peers, error) {
+	stakeReg, err := metrics.MakeAndRegister(
+		m.stakeGatherer,
+		primaryAlias,
+	)
+	if err != nil {
+		return nil, err
+	}
+	connectedValidators, err := tracker.NewMeteredPeers(stakeReg)
+	if err != nil {
+		return nil, fmt.Errorf("error creating peer tracker: %w", err)
+	}
+	return connectedValidators, nil
+}
+
+// createPeerTracker creates a peer tracker for the Snowman consensus engine
+func (m *manager) createPeerTracker(ctx *snow.ConsensusContext, primaryAlias string) (*p2p.PeerTracker, error) {
+	p2pReg, err := metrics.MakeAndRegister(
+		m.p2pGatherer,
+		primaryAlias,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	peerTracker, err := p2p.NewPeerTracker(
+		ctx.Log,
+		"peer_tracker",
+		p2pReg,
+		set.Of(ctx.NodeID),
+		nil,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("error creating peer tracker: %w", err)
+	}
+	return peerTracker, nil
+}
+
+func simplexCtxConfig(ctx *snow.ConsensusContext) simplex.SimplexChainContext {
```

Comment:
same, I think this should not be a function, it's only used once and is just an initialization of a struct.